server/src/main/java/org/elasticsearch/cluster/coordination/StableMasterHealthIndicatorService.java

```diff
@@ -162,14 +169,40 @@ private HealthIndicatorDetails getDetails(
                     coordinationDiagnosticsDetails.nodeToClusterFormationDescriptionMap()
                         .entrySet()
                         .stream()
-                        .map(entry -> Map.of("node_id", entry.getKey(), CLUSTER_FORMATION_MESSAGE, entry.getValue()))
+                        .map(
+                            entry -> Map.of(
+                                "node_id",
+                                entry.getKey(),
+                                "name",
+                                getNameForNodeId(entry.getKey()),
+                                CLUSTER_FORMATION_MESSAGE,
+                                entry.getValue()
+                            )
+                        )
                         .toList()
                 );
             }
             return builder.endObject();
         };
     }
 
+    /**
+     * Returns the name of the node with the given nodeId, as seen in the cluster state at this moment. The name of a node is optional,
+     * so if the node does not have a name (or the node with the given nodeId is no longer in the cluster state), an empty string is
+     * returned.
+     * @param nodeId The id of the node whose name is to be returned
+     * @return The current name of the node, or empty string if the node is not in the cluster state or does not have a name
+     */
+    private String getNameForNodeId(String nodeId) {
+        DiscoveryNode node = clusterService.state().nodes().get(nodeId);
+        if (node == null) {
+            return "";
```

Comment:
The ES APIs generally don't return empty fields, but rather omit the fields.

I think we should return null and omit the field in the response if it's null

Reply:
Makes sense.